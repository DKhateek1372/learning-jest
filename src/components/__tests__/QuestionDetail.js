import {mapStateToProps, QuestionDetailDisplay} from "../QuestionDetail";
import renderer from 'react-test-renderer';
import React from 'react';

describe('The Question Detail Component', () => {
  describe("The container", () => {
    describe("mapStateToProps", () => {
      it("should map state to props correctly", () => {

        const sampleQuestion = {
          question_id: 42,
          body: 'Space is big'
        };

        const appState = {
          questions: [sampleQuestion]
        };

        const ownProps = {
          question_id: 42
        };

        const componentState = mapStateToProps(appState, ownProps);

        expect(componentState).toEqual(sampleQuestion);

      });
    });
  });


  describe("The display", () => {
    it('Should not regress', () => {
      const tree = renderer.create(
        <QuestionDetailDisplay
          title="Hello"
          body="43"
          answer_count={2}
          tags={['stuff']}
        />
      );

      expect(tree.toJSON()).toMatchSnapshot();

    });
  });

});