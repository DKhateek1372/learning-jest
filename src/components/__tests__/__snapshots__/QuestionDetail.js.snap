// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component The display Should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Hello
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          stuff
        </code>
      </div>
    </div>
    <div>
      <p>
        43
      </p>
    </div>
    <div>
      2
       Answers
    </div>
  </div>
</div>
`;
